geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewayReceiverMeters.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.internal.cache.wan;
+
+import static java.lang.String.format;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+
+import org.apache.geode.Statistics;
+
+public class GatewayReceiverMeters {
+  private static final Set<String> meterWhitelist = new HashSet<String>() {
+    {
+      add("cache.gatewayreceiver.events.received");
```

Comment:
How is allow-listing going to match the full `Meter.Id` when you get into things like puts on specific regions? Seems to see we should be matching the entire `Meter.Id`. The probably being the `Meter.Id` is derived when constructing the Meter and so fetching it later from the `Meter` is too late. See my POC where I calculate the same `Meter.Id` prior to creating the new `Meter` and using that to match `Meter`s on registration.

Reply:
We didn't account for the possibility of selecting real/no-op meter based on tag values.

Can you give a scenario where always want a meter in the registry, but want to feed it to a stat only for certain regions?

I can think of a simpler scenario: For metadata regions, we may wish not to instrument certain operations at all. No meter in the registry. No updates to the stats. For that, the RegionMeters object can:
1. Construct a no-op meter without registering it.
2. Don't bother wiring the stat to it.

Then the instrumented code happily update the no-op meter.

If there's a scenario where we want to select real/no-op meter based on tag values, we'll have to make the accept list mutable, and remove `Meter.Id`s when the instrumented thing is deleted/closed/destroyed. And maybe some other stuff.

Reply:
Maybe in this model there would not be a need for matching on `Tag`.